src/evict/evict_page.c

```diff
@@ -692,6 +694,9 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     } else if (!WT_SESSION_BTREE_SYNC(session))
         LF_SET(WT_REC_VISIBLE_ALL);
 
+    if (checkpoint_running)
+        LF_SET(WT_REC_CHECKPOINT_RUNNING);
```

Comment:
I think you are repurposing that flag and I don't think that's the right approach. It should be OK for an eviction in a tree other than the one being checkpointed right now to evict content in front of the checkpoint snapshot. We did a lot of work to make that possible. If we disallow that all content will need to be kept in cache for the duration of the checkpoint.
The content being evicted in the tree currently being checkpointed is only evicted by the checkpoint thread - so the correct flag should already be set.

Reply:
I'm not repurposing the flag. It is a new flag I added in this change.

Reply:
bq. It should be OK for an eviction in a tree other than the one being checkpointed right now to evict content in front of the checkpoint snapshot. We did a lot of work to make that possible. If we disallow that all content will need to be kept in cache for the duration of the checkpoint.

I'm not holding up all the content to be evicted. The change only fails eviction if there is out of order timestamp updates in that page. If there is none, it can be evicted.

Reply:
I believe it is wrong to evict pages with out of order timestamp updates ahead of checkpoint as rollback to stable in this case doesn't see the original timestamp of the updates and may restore the wrong value.

Reply:
ack - my mistake. Thanks for explaining.